### PR TITLE
feat: dump and restore for graph objects

### PR DESCRIPTION
**Pull request by gtyun**

### CODE REVIEW

File: src/bin/pg_dump/pg_dump.c [hunk 1]
```diff
@@ -14685,6 +14702,103 @@ dumpDefaultACL(Archive *fout, DefaultACLInfo *daclinfo)
...
+						AND I.relname = '%s'",
+					fmtId(dobj->name));
+			res = ExecuteSqlQuery(fout, query->data, PGRES_TUPLES_OK);
+
+			if (PQntuples(res) == 1)
+				ret = true;
+			else
+				ret = false;
+			break;
+		case GO_Constraint :
```
Comment (htlim) [inline src/bin/pg_dump/pg_dump.c]: rename please.
This name seems to refer to all constraints.

File: src/bin/pg_dump/pg_dump.c [hunk 2]
```diff
@@ -14685,6 +14702,103 @@ dumpDefaultACL(Archive *fout, DefaultACLInfo *daclinfo)
...
+					AND c.relkind = 'S' \
+					AND c.relnamespace = g.nspid", fmtId(dobj->name));
+			res = ExecuteSqlQuery(fout, query->data, PGRES_TUPLES_OK);
+
+			if (PQntuples(res) == 1)
+				ret = true;
+			else
+				ret = false;
+			break;
+		case GO_Index :
```
Comment (htlim) [inline src/bin/pg_dump/pg_dump.c]: rename please.
This name seems to refer to all indexes of graph.

File: src/bin/pg_dump/pg_dump.c [hunk 3]
```diff
@@ -15361,9 +15475,34 @@ dumpTableSchema(Archive *fout, TableInfo *tbinfo)
...
+
+					if (IsGraphObject(fout, (DumpableObject *) tbinfo,
+										GO_Label, labkind))
+					{
+						if (strcmp(labkind, "v") == 0)
+							reltypename = "VLABEL";
+						else if (strcmp(labkind, "e") == 0)
+							reltypename = "ELABEL";
+						else
+							reltypename = "Invalid LABKIND";
```
Comment (htlim) [inline src/bin/pg_dump/pg_dump.c]: Is not this an error condition?

File: src/bin/pg_dump/pg_dump.c [hunk 4]
```diff
@@ -16440,6 +16584,15 @@ dumpSequence(Archive *fout, TableInfo *tbinfo)
 	PQExpBuffer delqry = createPQExpBuffer();
 	PQExpBuffer labelq = createPQExpBuffer();
 
+	if (IsGraphObject(fout, (DumpableObject *) tbinfo, GO_Sequence, NULL))
```
Comment (htlim) [inline src/bin/pg_dump/pg_dump.c]: The sequence created by the user in a graph is the target of the dump.

**Comment by gtyun** [comment]
rebase from latest master commit.

**Pull request closed without merging**